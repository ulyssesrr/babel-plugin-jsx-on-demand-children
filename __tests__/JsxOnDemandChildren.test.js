'use strict';

const transformerWithOptions = require('./transformerWithOptions');
const fs = require('fs');
const path = require('path');


/**
 * Extend Jest with a custom snapshot serializer to provide additional context
 * and reduce the amount of escaping that occurs.
 */
const FIXTURE_TAG = Symbol.for('FIXTURE_TAG');
expect.addSnapshotSerializer({
  print(value) {
    return Object.keys(value)
      .map(key => `~~~~~~~~~~ ${key.toUpperCase()} ~~~~~~~~~~\n${value[key]}`)
      .join('\n');
  },
  test(value) {
    return value && value[FIXTURE_TAG] === true;
  },
});


const fixturesPath = `${__dirname}/fixtures`
const fixtures = fs.readdirSync(fixturesPath);

test.each(fixtures)('matches expected output: %s', file => {
  const input = fs.readFileSync(path.join(fixturesPath, file), 'utf8');
  
  const output = transformerWithOptions({})(input, file);

  expect({
    [FIXTURE_TAG]: true,
    input,
    output,
  }).toMatchSnapshot();
});