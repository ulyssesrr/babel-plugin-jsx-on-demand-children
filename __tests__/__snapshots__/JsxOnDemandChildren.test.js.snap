// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches expected output: invalid-child.txt 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
'use strict';

import React from 'react';

function Test() {
  return (
    <Parent>
      <JsxOnDemandChildren>
        <LeafChild prop={true} />
        {invalidChild}
      </JsxOnDemandChildren>
    </Parent>
  );
};
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
'use strict';

import React from 'react';

function Test() {
  return (
    <Parent>
      <JsxOnDemandChildren>
        <LeafChild prop={true} />
        {invalidChild}
      </JsxOnDemandChildren>
    </Parent>
  );
}

`;

exports[`matches expected output: simple.txt 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
'use strict';

import React from 'react';

function Test() {
  return (
    <Parent>
      <JsxOnDemandChildren>
        <LeafChild prop={true} />
      </JsxOnDemandChildren>
    </Parent>
  );
};
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
'use strict';

import React from 'react';

function Test() {
  return <Parent LeafChild={() => <LeafChild prop={true} />}></Parent>;
}

`;

exports[`matches expected output: simple-other-children.txt 1`] = `
~~~~~~~~~~ INPUT ~~~~~~~~~~
'use strict';

import React from 'react';

function Test() {
  return (
    <Parent>
      <JsxOnDemandChildren>
        <LeafChild prop={true} />
      </JsxOnDemandChildren>
      <Other />
    </Parent>
  );
};
~~~~~~~~~~ OUTPUT ~~~~~~~~~~
'use strict';

import React from 'react';

function Test() {
  return (
    <Parent LeafChild={() => <LeafChild prop={true} />}>
      <Other />
    </Parent>
  );
}

`;
